src/onegov/agency/forms/agency.py

```diff
@@ -171,13 +171,20 @@ def update_model(self, model):
         session = self.request.session
         agencies = ExtendedAgencyCollection(session)
 
-        parent_id = None
-        parent = None
+        new_parent_id = None
+        new_parent = None
         if self.parent_id.data and self.parent_id.data.isdigit():
-            parent_id = int(self.parent_id.data)
-            parent = agencies.by_id(parent_id)
-        model.name = agencies.get_unique_child_name(model.title, parent)
-        model.parent_id = parent_id
+            new_parent_id = int(self.parent_id.data)
+            new_parent = agencies.by_id(new_parent_id)
+
+        # prevent assigning yourself as parent
+        if model.id != new_parent_id:
```

Comment:
Error handling should be done in the form, either using a validator, a ensure_ or validate_ function, e.g. https://github.com/OneGov/onegov-cloud/blob/master/src/onegov/feriennet/forms/userprofile.py#L181

Reply:
will rework it

Reply:
done